### PR TITLE
feat: dia-862: support for cancel model run

### PR DESCRIPTION
**Pull request by forum-hs**

### CODE REVIEW

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/4a5fb04964fdb7c7e9337cadd3b2d6443477bfd3 is created.

File: server/app.py [hunk 1]
```diff
@@ -324,6 +324,12 @@ def cancel_job(job_id):
         JobStatusResponse[status.CANCELED]
     """
     job = process_file.AsyncResult(job_id)
+    input_job_id = job.info["input_job_id"]
```
Comment (hakan458) [inline server/app.py:327]: Add checks to make sure this metadata exists, for both input and output

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/f9f438d4a37146d908c6bc5a4e248de3892145b3 is created.

File: server/app.py (hunk 1 above)
Comment (forum-hs) [reply server/app.py:327]: it should right. on would it stop existing if the job is done?

Comment (forum-hs) [reply server/app.py]: added the check

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/8ebfe18186213e2a4412eb097bfeec07ccfc5751 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/151a7db6c45110f10c67dd90257fb999ed5c6bf0 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/cd75157c03b5ef55027271fefe04acd89c509573 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/350efe51d446470e7e3fac10f3017e46ba7374be is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/cb1b096b6b1471a621117d0819172888f16dc4f6 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/2ee333775462806607d28af3d1e8d54335f04156 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/74e42c8d8aa964e45ec95a1e11a345ebbef8a8b0 is created.

File: server/app.py [hunk 2]
```diff
@@ -324,6 +324,21 @@ def cancel_job(job_id):
         JobStatusResponse[status.CANCELED]
     """
     job = process_file.AsyncResult(job_id)
```
Comment (matt-bernstein) [inline server/app.py:326]: should be `streaming_parent_task.AsyncResult`

File: server/app.py [hunk 3]
```diff
@@ -324,6 +324,21 @@ def cancel_job(job_id):
...
+    input_job_id = job.info.get("input_job_id", None)
+    output_job_id = job.info.get("output_job_id")
+    job_flag = {}
```
Comment (matt-bernstein) [inline server/app.py:329]: `job_flag` is unused?

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/f151ba54b7e11d792e9daecedc34fe85f1ecd10a is created.

File: server/app.py [hunk 4]
```diff
@@ -324,6 +324,21 @@ def cancel_job(job_id):
         JobStatusResponse[status.CANCELED]
     """
     job = process_file.AsyncResult(job_id)
+    input_job_id = job.info.get("input_job_id", None)
+    output_job_id = job.info.get("output_job_id")
```
Comment (matt-bernstein) [inline server/app.py:328]: output_job_id = job.info.get("output_job_id", None)

**Review by matt-bernstein — APPROVED** [review]

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/7cb0043fe40b35b71203ac7a3531fe3b5b165cea is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/77bcbac0ab3487e046361a22149fddc97d84e225 is created.

**Comment by robot-ci-heartex** [comment]
Commit https://github.com/HumanSignal/infra/commit/c39b8d49a7222f6164f026376884fbf3ad5d6da3 is created.

**Pull request merged**